apps/aechannel/src/aesc_fsm.erl

```diff
@@ -461,6 +484,30 @@ strict_checks(Fsm, Strict) when is_boolean(Strict) ->
     gen_statem:call(Fsm, {strict_checks, Strict}).
 -endif.
 
+%% ======================================================================
+%% FSM count and limit
+
+register_aggregated_counter() ->
+    gproc:reg({a,l,?MODULE}),
+    lager:debug("Aggregated counter registered: value = ~p", [number_of_fsms()]),
+    ok.
+
+register_counter() ->
+    gproc:reg({c,l,?MODULE}, 1),
+    lager:debug("counter registered: current count: ~p", [number_of_fsms()]),
+    ok.
+
+number_of_fsms() ->
+    gproc:lookup_value({a,l,?MODULE}).
+
+within_fsm_number_limit() ->
+    {ok, Max} = aeu_env:find_config([ <<"channels">>, <<"max_count">> ] , [ user_config
+                                                                          , schema_default
+                                                                          , {value, 1000} ]),
+    N = number_of_fsms(),
+    lager:debug("Number of fsms: ~p; Max = ~p", [N, Max]),
+    N < Max.
```

Comment:
Here `N` does not include the include the FSM being started at the moment so the maximum amount of FSMs is indeed according to [description](https://github.com/aeternity/aeternity/pull/2805/files#diff-fc6905deb828c517288fba567781e76fR323). Ok